### PR TITLE
23.12 release updates

### LINKED ISSUE #441: [DOC] 23.12 CUDA 12.0 Support Updates
CUDA 12.0 on ARM is now supported through `23.12` conda and Docker nightlys and https://github.com/rapidsai/docs/pull/440 made the install page/selector reflect this.

With the 23.12 release PRs we will need to include changes to remove the caveats about CUDA 12.0 on ARM at all since they aren't relevant anymore.

### PR DESCRIPTION
**Pull request by jarmak-nv**
This PR does 3 things:
- Standard updates for a new release
- Closes all cuSignal doc links, but still references it and points to cuPy
- Close #441 

cc @caryr35 @raydouglass

### CODE REVIEW

File: _data/releases.json [hunk 1]
```diff
@@ -82,5 +44,43 @@
       "end": "Thu Feb 8 2024",
       "days": "2"
     }
+  },
+  "next_nightly": {
+    "version": "24.04",
+    "cudf_dev": {
+      "start": "1 19 2024",
```
Comment (bdice) [inline _data/releases.json:51]: I think the other dates use month abbreviations.
```suggestion
      "start": "Jan 19 2024",
```

**Review by bdice — APPROVED** [review]
Approving for correctness - but I am not 100% sure about whether this is entirely complete. I think there’s some other cleanup I had in mind for 23.12 but I don’t remember what it is right now.

**Comment by jarmak-nv** [comment]
> Approving for correctness - but I am not 100% sure about whether this is entirely complete. I think there’s some other cleanup I had in mind for 23.12 but I don’t remember what it is right now.

Happy to make any other changes; maybe it's the supported CUDA versions?
https://github.com/rapidsai/docs/blob/8cd7b14d15a278e82c35006c9abc2e50af92c7a5/install/install.md?plain=1#L134-L138

I think technically any of those will work if you use a `-cu11` wheel, but for supported I think it's just 11.4/8 + 12.0, right?

File: _data/docs.yml [hunk 2]
```diff
@@ -82,12 +82,12 @@ apis:
   cusignal:
     name: cusignal
     path: cusignal
-    desc: 'cuSignal leverages CuPy, Numba, and the RAPIDS ecosystem for GPU accelerated signal processing. In some cases, cuSignal is a direct port of Scipy Signal to leverage GPU compute resources via CuPy but also contains Numba CUDA kernels for additional speedups for selected functions.'
-    ghlink: https://github.com/rapidsai/cusignal
-    cllink: https://github.com/rapidsai/cusignal/blob/main/CHANGELOG.md
+    desc: 'cuSignal functionality  has been moved to CuPy. Please see the CuPy documentation for more information.'
```
Comment (bdice) [inline _data/docs.yml:85]:
```suggestion
    desc: 'cuSignal functionality has been moved to CuPy. Please see the CuPy documentation for more information.'
```

**Review by raydouglass — APPROVED** [review]
Also asked [here](https://github.com/rapidsai/rapids.ai/pull/349#issuecomment-1852267814), but should we add the version for stable pip commands?

This would help prevent [install errors](https://github.com/rapidsai/cuml/issues/5237#issuecomment-1850205870) during a release.

**Comment by bdice** [comment]
> No objection to adding release versions - do we need tertiary versions or will secondary suffice since pip _should_ pick up the most recent hotfix, right? (ie 23.12 vs 23.12.0)

We need `==23.12.*` to pick up patch releases.

**Comment by jarmak-nv** [comment]
Pinning is done, I dropped it down to 3 packages per line for stable releases from 5 to make it more readable as we've added a bunch of text

**Comment by jarmak-nv** [comment]
Oh right I need to update the cucim install command for stable as well now that it will be a `-cu11/12` wheel like everything else

**Pull request merged**